recipe_Menager.py

```diff
@@ -2,26 +2,105 @@ class RecipeManager:
+        for instruction in instructions:
+            new_recipe.set_instructions(instruction.strip())
+        
+        self.add_recipe(new_recipe)
+        print("Recipe added.")
+        
+
+    def update_recipe(self, recipe_to_update: Recipe, updated_recipe: Recipe):
+        recipe_to_update.set_description(updated_recipe.get_description())
+        recipe_to_update.set_no_servings(updated_recipe.get_no_servings())
+        recipe_to_update.set_calories_per_portion(updated_recipe.get_calories_per_portion())
+        recipe_to_update.set_ingredients(updated_recipe.get_ingredients())
+        recipe_to_update.set_instructions(updated_recipe.get_instructions())
+    def update_recipe_from_input(self):
+        title_to_update = input("Enter the name of the recipe to update: ")
+        found = False
+        for recipe in self.recipes:
+            if recipe._title == title_to_update:
+                description = input("Enter the updated description: ")
+                no_servings = int(input("Enter the updated number of servings: "))
+                calories_per_portion = int(input("Enter the updated calories per portion: "))
+                ingredients = input("Enter the updated ingredients: ").split(",")
+                instructions = input("Enter the updated instructions: ").split(",")
+                updated_recipe = Recipe(title_to_update)
+                updated_recipe.set_description(description)
+                updated_recipe.set_no_servings(no_servings)
+                updated_recipe.set_calories_per_portion(calories_per_portion)
+                for ingredient in ingredients:
+                    updated_recipe.set_ingredients(ingredient.strip())
+                for instruction in instructions:
+                    updated_recipe.set_instructions(instruction.strip())
+
+                self.update_recipe(recipe, updated_recipe)
+                print("Recipe updated successfully.")
+                found = True
+                break
+
+        if not found:
+            print("Recipe not found.")
+        
 
-    @display_recipes.setter
-    def add_recipe(self, recipe):
-        self.recipes.append(recipe)
 
-    def update_recipe(self, recipe_index, new_recipe):
-        self.recipes[recipe_index] = new_recipe
     
-    def delete_recipe(self, recipe_index):
-        self.recipes.pop(recipe_index)
+    def delete_recipe(self, recipe_to_delete: Recipe):
+        self.recipes.remove(recipe_to_delete)
+    def delete_recipe_from_input(self):
+        title_to_delete = input("Enter the name of the recipe to delete: ")
+        found = False
+
+        for recipe in self.recipes:
+            if recipe._title == title_to_delete:
+                self.delete_recipe(recipe)
+                print("Recipe deleted successfully.")
+                found = True
+                break
 
+        if not found:
+            print("Recipe not found. Unable to delete.")
 
 
+    def menu(self):
```

Comment:
Again useful for debug/development/test but needs to be either in the test file or contained in a derived test class.